src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -748,7 +748,11 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                             ref optionAssuredNonBlockingTrue,
                             (uint)Marshal.SizeOf<uint>()))
                         {
-                            WinHttpException.ThrowExceptionUsingLastError();
+                            int lastError = Marshal.GetLastWin32Error();
+                            if (lastError != Interop.WinHttp.ERROR_WINHTTP_INVALID_OPTION)
```

Comment:
How often do we create a new session handle?  I'm wondering if there's performance value in caching into a static boolean a false if we get this error, and then only try to set this option subsequently if that boolean is still true.